src/common/stacktrace.rs

```diff
@@ -0,0 +1,86 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, JsonSchema, Debug)]
+struct StackTraceSymbol {
+    name: Option<String>,
+    file: Option<String>,
+    line: Option<u32>,
+}
+
+#[derive(Deserialize, Serialize, JsonSchema, Debug)]
+struct StackTraceFrame {
+    symbols: Vec<StackTraceSymbol>,
+}
+
+impl StackTraceFrame {
+    pub fn render(&self) -> String {
+        let mut result = String::new();
+        for symbol in &self.symbols {
+            let symbol_string = format!(
+                "{}:{} - {} ",
```

Comment:
```suggestion
                "{}:{} - {}",
```

Reply:
that's on purpose, as there a list of symbols